tests/integration/api/test_api.py

```diff
@@ -521,221 +357,156 @@ def _compare_screenboard(apiBoard, expectedBoard):
-        eq(dt['message'], message)
-        eq(dt['disabled'], False)
+        downtime = dog.Downtime.update(downtime["id"], scope="env:staging", end=end, message=message)
+        assert downtime["end"] == end
+        assert downtime["message"] == message
+        assert downtime["disabled"] is False
 
         # Delete downtime
-        dog.Downtime.delete(downtime_id)
-        dt = dog.Downtime.get(downtime_id)
-        eq(dt['disabled'], True)
+        assert dog.Downtime.delete(downtime["id"]) is None
+        downtime = dog.Downtime.get(downtime["id"])
+        assert downtime["disabled"] is True
 
-    @pytest.mark.monitor
     def test_service_check(self):
-        dog.ServiceCheck.check(
-            check='check_pg', host_name='host0', status=1,
-            message='PG is WARNING', tags=['db:prod_data'])
+        assert dog.ServiceCheck.check(
+            check="check_pg",
+            host_name="host0",
+            status=1,
+            message="PG is WARNING",
+            tags=["db:prod_data"]
+        ) == {"status": "ok"}
 
-    @pytest.mark.host
     def test_host_muting(self):
-        hostname = "my.test.host"
-        message = "Muting this host for a test."
         end = int(time.time()) + 60 * 60
+        hostname = "my.test.host" + str(end)
+        message = "Muting this host for a test."
 
-        try:
-            # reset test
-            dog.Host.unmute(hostname)
-        except ApiError:
-            pass
+        # post a metric to make sure the test host context exists
+        dog.Metric.send(metric="test.muting.host", points=1, host=hostname)
+        # Wait for host to appear
+        get_with_retry("Tag", hostname)
 
         # Mute a host
         mute = dog.Host.mute(hostname, end=end, message=message)
-        eq(mute['hostname'], hostname)
-        eq(mute['action'], "Muted")
-        eq(mute['message'], message)
-        eq(mute['end'], end)
+        assert mute["hostname"] == hostname
+        assert mute["action"] == "Muted"
+        assert mute["message"] == message
+        assert mute["end"] == end
 
-        # We shouldn't be able to mute a host that's already muted, unless we include
+        # We shouldn"t be able to mute a host that"s already muted, unless we include
         # the override param.
         end2 = end + 60 * 15
-        resp = dog.Host.mute(hostname, end=end2)
-        assert resp['errors'][0].startswith('host:my.test.host is already muted.')
-
+        assert "errors" in dog.Host.mute(hostname, end=end2)
         mute = dog.Host.mute(hostname, end=end2, override=True)
-        eq(mute['hostname'], hostname)
-        eq(mute['action'], "Muted")
-        eq(mute['end'], end2)
+        assert mute["hostname"] == hostname
+        assert mute["action"] == "Muted"
+        assert mute["end"] == end2
 
-        dog.Host.unmute(hostname)
+        unmute = dog.Host.unmute(hostname)
+        assert unmute["hostname"] == hostname
+        assert unmute["action"] == "Unmuted"
```

Comment:
Should we get with retry and assert the host is muted here, or should this be enough?

Reply:
This is enough. We make a call and the API returns something to us that we can assert on. trying to get again would be testing that the backend correctly does its job, which is not the goal here

Reply:
Ok thats fair. The `get` itself is tested elsewhere, so that makes sense.